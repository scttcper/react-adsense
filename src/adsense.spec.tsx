import React from 'react';
import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import { Adsense } from './index';

describe('Adsense', () => {
  it('should be instantiable', () => {
    const { container } = render(<Adsense client="123" slot="abc" className="ads-class" />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it('should allow children', () => {
    const { container } = render(
      <Adsense client="123" slot="abc" className="ads-class">
        <img alt="placeholder" />
      </Adsense>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it('should call window adsbygoogle', () => {
    (window as any).adsbygoogle = [];
    render(<Adsense client="123" slot="abc" />);
    expect((window as any).adsbygoogle).toHaveLength(1);
    (window as any).adsbygoogle = undefined;
  });
});
