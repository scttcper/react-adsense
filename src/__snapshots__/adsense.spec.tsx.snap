// Vitest Snapshot v1

exports[`Adsense > should allow children 1`] = `
<ins
  class="adsbygoogle ads-class"
  data-ad-client="123"
  data-ad-format="auto"
  data-ad-layout=""
  data-ad-layout-key=""
  data-ad-slot="abc"
  data-full-width-responsive="false"
  style="display: block;"
>
  <img
    alt="placeholder"
  />
</ins>
`;

exports[`Adsense > should be instantiable 1`] = `
<ins
  class="adsbygoogle ads-class"
  data-ad-client="123"
  data-ad-format="auto"
  data-ad-layout=""
  data-ad-layout-key=""
  data-ad-slot="abc"
  data-full-width-responsive="false"
  style="display: block;"
/>
`;
